### PR TITLE
Add warning for even replication factor

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Ass discussed in slack https://camunda.slack.com/archives/C037RS2JHB8/p1677221646815649 I added a warning for even replication factor, as this is also a question or misconfiguration I often have seen in the forum.

I would like to back port this to our other supported versions.

### CODE REVIEW

File: broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java [hunk 1]
```diff
@@ -40,6 +40,12 @@ public final class SystemContext {
       "Node id %s needs to be non negative and smaller then cluster size %s.";
   private static final String REPLICATION_FACTOR_ERROR_MSG =
       "Replication factor %s needs to be larger then zero and not larger then cluster size %s.";
+  private static final String REPLICATION_FACTOR_WARN_MSG =
+      "Expected to have odd replication factor, but was even ({}). Even replication factor has no benefit over "
+          + "the next odd value. Quorum is calculated as:"
```
Comment (oleschoenburg) [inline broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java:45]:
```suggestion
          + "the previous odd value. Quorum is calculated as:"
```

Comment (Zelldon) [reply broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java:45]: Actually I meant next, because the quorum value will be the same.

**Comment by Zelldon** [comment]
@oleschoenburg can you take a look again.

File: broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java (hunk 1 above)
Comment (oleschoenburg) [reply broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java:45]: I understand but I think it's still correct and important to say: "replication factor 2 is not better than 1 and it's worse than 3, in most cases you should use 1 or 3"

If we only call out that 2 is not better than 3 ("Even replication factor has no benefit over the next odd value"), it sounds like it is just as good as 3 where it actually is worse.

**Review by lenaschoenburg — APPROVED** [review]

File: broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java (hunk 1 above)
Comment (Zelldon) [reply broker/src/main/java/io/camunda/zeebe/broker/system/SystemContext.java]: Thanks @oleschoenburg for the feedback I adjusted the warning.

**Pull request merged**